### PR TITLE
Feature show recommendations

### LINKED ISSUE #18: Feature show recommendations
As a user I would like to see my recommendations.

### PR DESCRIPTION
**Pull request by icerrate**
Close #18

### CODE REVIEW

File: app/src/main/java/com/belatrixsf/allstars/adapters/EmployeeListAdapter.java [hunk 1]
```diff
@@ -25,16 +40,14 @@
 public class EmployeeListAdapter extends RecyclerView.Adapter<EmployeeListAdapter.EmployeeViewHolder> {
 
     private List<Employee> employeeList;
-    private WeakReference<Context> context;
-    private String lastFirstLetter;
+    private List<String> letters = new ArrayList<>();
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/allstars/adapters/EmployeeListAdapter.java]: Not used

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -26,7 +26,5 @@
     <dimen name="text_28">28sp</dimen>
 
     <dimen name="corner_radius">4dp</dimen>
-    <dimen name="picture_margin_size">20dp</dimen>
```
Comment (icerrate) [inline app/src/main/res/values/dimens.xml:29]: Are required
-picture_margin_size
-picture_stroke_width

File: app/src/main/java/com/belatrixsf/allstars/adapters/RecommendationListAdapter.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    @Override
+    public RecommendationViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View layoutView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_folding_recommendation, parent, false);
+        return new RecommendationViewHolder(layoutView);
+    }
+
+    @Override
+    public void onBindViewHolder(final RecommendationViewHolder holder, int position) {
+        Recommendation recommendation = recommendationList.get(position);
+        holder.fc.setOnClickListener(new View.OnClickListener() {
```
Comment (cpinan) [inline app/src/main/java/com/belatrixsf/allstars/adapters/RecommendationListAdapter.java]: This onClick should be moved into the ViewHolder

File: app/src/main/java/com/belatrixsf/allstars/services/EmployeeServerService.java [hunk 4]
```diff
@@ -61,6 +60,10 @@ public void getEmployeeList(AllStarsCallback<SearchEmployeeResponse> callback) {
     @Override
     public void getEmployeeSearchList(String searchTerm, AllStarsCallback<SearchEmployeeResponse> callback) {
         employeeAPI.getEmployeeSearchList(searchTerm).enqueue(new RetrofitCallback<SearchEmployeeResponse>(callback));
+    }
 
+    @Override
+    public void getRecommendationList(int employeeId, int subcateogryId, AllStarsCallback<RecommendationResponse> callback) {
```
Comment (cpinan) [inline app/src/main/java/com/belatrixsf/allstars/services/EmployeeServerService.java]: Misspelled should be "subcategoryId"

File: app/src/main/java/com/belatrixsf/allstars/ui/recommendation/RecommendationPresenter.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    private EmployeeService employeeService;
+
+    @Inject
+    public RecommendationPresenter(RecommendationView view, EmployeeService employeeService) {
+        super(view);
+        this.employeeService = employeeService;
+    }
+
+    public void getRecommendationList(int employeeId, int subcategoryId) {
+        employeeService.getRecommendationList(employeeId, subcategoryId, new AllStarsCallback<RecommendationResponse>() {
```
Comment (cpinan) [inline app/src/main/java/com/belatrixsf/allstars/ui/recommendation/RecommendationPresenter.java:45]: I think we should cancel the request if the device goes to background mode, for example, a call enters or the devices is locked.

File: app/src/main/res/layout/item_folding_recommendation.xml [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+                    android:layout_marginTop="@dimen/activity_vertical_margin">
+
+                    <TextView
+                        android:id="@+id/content_label_user"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_alignParentLeft="true"
+                        android:textStyle="bold"
+                        android:layout_alignParentTop="true"
+                        android:text="From:" />
```
Comment (cpinan) [inline app/src/main/res/layout/item_folding_recommendation.xml]: Move from to strings.xml

File: app/src/main/res/layout/item_folding_recommendation.xml [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+
+                    <TextView
+                        android:id="@+id/content_user"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_alignParentRight="true"
+                        android:layout_alignParentTop="true"
+                        tools:text="Name" />
```
Comment (cpinan) [inline app/src/main/res/layout/item_folding_recommendation.xml]: Move from to strings.xml

File: app/src/main/res/layout/item_folding_recommendation.xml [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+
+                    <TextView
+                        android:id="@+id/content_label_message"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_alignParentLeft="true"
+                        android:textStyle="bold"
+                        android:layout_marginTop="@dimen/text_margin"
+                        android:layout_below="@+id/content_user"
+                        android:text="Message:" />
```
Comment (cpinan) [inline app/src/main/res/layout/item_folding_recommendation.xml]: Move from to strings.xml

File: app/src/main/res/layout/item_folding_recommendation.xml [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+                    <TextView
+                        android:id="@+id/content_message"
+                        android:layout_width="match_parent"
+                        android:layout_height="wrap_content"
+                        android:textStyle="italic"
+                        android:layout_below="@+id/content_label_message"
+                        tools:text="Text" />
```
Comment (cpinan) [inline app/src/main/res/layout/item_folding_recommendation.xml]: Move from to strings.xml

File: app/src/main/res/layout/item_folding_recommendation.xml [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+
+                    <TextView
+                        android:id="@+id/content_label_category"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_alignParentLeft="true"
+                        android:textStyle="bold"
+                        android:layout_marginTop="@dimen/text_margin"
+                        android:layout_below="@+id/content_message"
+                        android:text="Category:" />
```
Comment (cpinan) [inline app/src/main/res/layout/item_folding_recommendation.xml]: Move from to strings.xml

File: app/src/main/res/layout/item_folding_recommendation.xml [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+
+                    <TextView
+                        android:id="@+id/content_category"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_marginTop="@dimen/text_margin"
+                        android:layout_alignParentRight="true"
+                        android:layout_below="@+id/content_message"
+                        tools:text="Text" />
```
Comment (cpinan) [inline app/src/main/res/layout/item_folding_recommendation.xml]: Move from to strings.xml

File: app/src/main/res/layout/item_folding_recommendation.xml [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+
+                    <TextView
+                        android:id="@+id/content_label_date"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_alignParentLeft="true"
+                        android:textStyle="bold"
+                        android:layout_marginTop="@dimen/text_margin"
+                        android:layout_below="@+id/content_category"
+                        android:text="Date:" />
```
Comment (cpinan) [inline app/src/main/res/layout/item_folding_recommendation.xml]: Move from to strings.xml

File: app/src/main/res/layout/item_folding_recommendation.xml [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+
+                    <TextView
+                        android:id="@+id/content_date"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_alignParentRight="true"
+                        android:layout_marginTop="@dimen/text_margin"
+                        android:layout_below="@+id/content_category"
+                        tools:text="Text" />
```
Comment (cpinan) [inline app/src/main/res/layout/item_folding_recommendation.xml]: Move from to strings.xml

**Comment by cpinan** [comment]
Changes merged into develop branch.

**Pull request merged**